### PR TITLE
Add support to change the SE context window of ConvASREncoder

### PR DESCRIPTION
**Pull request by titu1994**
# Changelog
- Adds the support to change the SqueezeExcite context window post instantiation inside a given ConvASREncoder module.
- Adds tests to ensure the model updates the configs and changes the modules SE behaviour

# Usage

```python
model = ASRModel.from_pretrained("stt_en_citrinet_256")
model.change_conv_asr_se_context_window(context_window=128, update_config=True)  # equivalent to 128 * 0.01s of context
```

### CODE REVIEW

**Review by blisc — APPROVED** [review]

File: nemo/collections/asr/parts/asr_module_utils.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+def _update_se_context_window(model: 'ASRModel', context_window: int, cfg: Optional[DictConfig] = None):
+    jasper_block_counter = -1
+    for name, m in model.named_modules():
+        if type(m) == jasper.JasperBlock:
+            jasper_block_counter += 1
+
+        if type(m) == jasper.MaskedConv1d:
+            if m.conv.stride[0] > 1 and 'mconv' in name:
+                context_window = context_window // 2
```
[the remark this answers is not in the document]
Comment (titu1994) [reply nemo/collections/asr/parts/asr_module_utils.py]: Done

File: nemo/collections/asr/models/ctc_models.py [hunk 2]
```diff
@@ -36,7 +37,7 @@
 __all__ = ['EncDecCTCModel']
 
 
-class EncDecCTCModel(ASRModel, ExportableEncDecModel):
+class EncDecCTCModel(ASRModel, ExportableEncDecModel, ASRModuleMixin):
```
Comment (VahidooX) [inline nemo/collections/asr/models/ctc_models.py:40]: Shouldn't it be ASRModelMixin?

File: nemo/collections/asr/parts/asr_module_utils.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            _update_se_context_window(model, context_window, cfg=enc_cfg)
+    else:
+        _update_se_context_window(model, context_window)
+
+    # Update model config
+    if update_config:
+        model.cfg.encoder = enc_cfg
+
+
+def _update_se_context_window(model: 'ASRModel', context_window: int, cfg: Optional[DictConfig] = None):
```
Comment (VahidooX) [inline nemo/collections/asr/parts/asr_module_utils.py:67]: What happens if user calls it on a Conformer?

File: nemo/collections/asr/parts/asr_module_utils.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+from typing import Optional
+
+from omegaconf import DictConfig, open_dict
+
+from nemo.collections.asr.modules import conv_asr
+from nemo.collections.asr.parts import jasper
+from nemo.utils import logging
+
+
+def change_conv_asr_se_context_window(model: 'ASRModel', context_window: int, update_config: bool = True):
```
Comment (VahidooX) [inline nemo/collections/asr/parts/asr_module_utils.py:24]: This method always need a model as the input, how about moving it into the ASRModuleMixin as user is supposed to call it via the model?

**Review by VahidooX — APPROVED** [review]

**Pull request merged**